src/parser/calculate_size.ts

```diff
@@ -77,9 +78,19 @@ function calculateElement(
     case 'boolean':
       return (name != null ? ByteUtils.utf8ByteLength(name) + 1 : 0) + (1 + 1);
     case 'object':
-      if (value == null || value['_bsontype'] === 'MinKey' || value['_bsontype'] === 'MaxKey') {
+      if (
+        value != null &&
+        typeof value._bsontype === 'string' &&
+        value[Symbol.for('@@mdb.bson.version')] !== constants.BSON_MAJOR_VERSION
+      ) {
+        throw new BSONError('Unsupported BSON version, bson types must be from bson 5.0 or later');
```

Comment:
I've seen discussion about creating an error hierarchy for BSON.

Could we consider starting that now?  This seems like a good candidate for a specialized error, `BSONTypeVersionMismatchError`.

Reply:
Done